src/snowflake/snowpark/_internal/analyzer/snowflake_plan_node.py

```diff
@@ -32,6 +32,7 @@ class LogicalPlan:
     def __init__(self) -> None:
         self.children = []
```

Comment:
you also need to add update in the binary_plan_node.py to update the left and right correclty.

Reply:
moved to utils.py and made the change there.